client/apps.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+type AppInfo struct {
+	Snap         string `json:"snap,omitempty"`
+	Name         string `json:"name"`
+	DesktopFile  string `json:"desktop-file,omitempty"`
+	*ServiceInfo `json:",omitempty"`
+}
+
+// IsService returns true if the application is a background daemon.
+func (a *AppInfo) IsService() bool {
+	if a == nil {
+		return false
+	}
+	if a.ServiceInfo == nil {
+		return false
+	}
+	if a.ServiceInfo.Daemon == "" {
+		return false
+	}
+
+	return true
+}
+
+type ServiceInfo struct {
+	Daemon          string `json:"daemon"`
+	ServiceFileName string `json:"service-file-name"`
+	Enabled         bool   `json:"enabled"`
+	Active          bool   `json:"active"`
```

Comment:
Actually, sorry. Right after sending this comment my brain turned around and realized that the fact we use this for filtering doesn't necessarily mean it's a class of application. It might simply mean "it has a command today", much like Refresh simply means "it has a refresh available" in FindOptions.

The latter point in the details above then becomes more relevant: we need to decide first whether we want to enable daemons as commands in the local system. If we do that, then filtering by commands becomes unnecessary as everything has a command.

As such, can we please just take that specific filtering off right now and filter on the client side for the time being (on Daemon)?  We can reintroduce it later if we decide not to go with services with commands.